homeassistant/components/plugwise/climate.py

```diff
@@ -0,0 +1,310 @@
+        """Return the active preset mode."""
+        return self._preset_mode
+
+    @property
+    def preset_modes(self):
+        """Return the available preset modes list."""
+        return self._preset_modes
+
+    @property
+    def hvac_modes(self):
+        """Return the available hvac  modes list."""
+        return self._hvac_modes
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def min_temp(self):
+        """Return the minimal temperature possible to set."""
+        return self._min_temp
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature possible to set."""
+        return self._max_temp
+
+    @property
+    def target_temperature(self):
+        """Return the target temperature."""
+        return self._temperature
+
+    @property
+    def outdoor_temperature(self):
+        """Return the outdoor temperature."""
+        return self._outdoor_temperature
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measured temperature."""
+        return TEMP_CELSIUS
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        _LOGGER.debug("Adjusting temperature")
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if (temperature is not None and self._min_temp < temperature
+                < self._max_temp):
+            self._temperature = temperature
+            domain_objects = self._api.get_domain_objects()
+            self._api.set_temperature(domain_objects, temperature)
+            self.schedule_update_ha_state()
+            _LOGGER.debug("Changing temporary temperature")
+        else:
+            _LOGGER.error("Invalid temperature requested")
+
+    def set_hvac_mode(self, hvac_mode):
+        """Set the hvac mode."""
+        _LOGGER.debug("Adjusting hvac_mode (i.e. schedule/schema)")
+        if self._previous_schema is None and self._active_schema is None:
+            _LOGGER.error("previous_schema not known, unable to apply")
+        else:
+            if self._previous_schema is None:
+                schema = self._active_schema
+            else:
+                schema = self._previous_schema
+            if hvac_mode == 'auto':
+                self._hvac_mode = HVAC_MODE_AUTO
+                schema_mode = 'true'
+            else:
+                self._hvac_mode = HVAC_MODE_OFF
+                schema_mode = 'false'
+            domain_objects = self._api.get_domain_objects()
+            set_schema = self._api.set_schema_state(domain_objects,
+                                                    schema, schema_mode)
+            return set_schema
+
+    def set_preset_mode(self, preset_mode):
+        """Set the preset mode."""
+        _LOGGER.debug("Adjusting preset_mode (i.e. preset)")
+        if preset_mode is not None and preset_mode in self._preset_modes:
```

Comment:
None check is not needed. None is not in the list.

Reply:
Removed from the statement